mailbox/elasticsearch-v6/src/main/java/org/apache/james/mailbox/elasticsearch/v6/search/ElasticSearchSearcher.java

```diff
@@ -19,101 +19,102 @@
 
-    public ElasticSearchSearcher(Client client, QueryConverter queryConverter, int size,
+    public ElasticSearchSearcher(RestHighLevelClient client, QueryConverter queryConverter, int size,
                                  MailboxId.Factory mailboxIdFactory, MessageId.Factory messageIdFactory,
-                                 ReadAliasName aliasName, TypeName typeName) {
+                                 ReadAliasName aliasName) {
         this.client = client;
         this.queryConverter = queryConverter;
         this.size = size;
         this.mailboxIdFactory = mailboxIdFactory;
         this.messageIdFactory = messageIdFactory;
         this.aliasName = aliasName;
-        this.typeName = typeName;
     }
 
     public Stream<MessageSearchIndex.SearchResult> search(Collection<MailboxId> mailboxIds, SearchQuery query,
-                                                          Optional<Long> limit) throws MailboxException {
-        SearchRequestBuilder searchRequestBuilder = getSearchRequestBuilder(client, mailboxIds, query, limit);
-        Stream<MessageSearchIndex.SearchResult> pairStream = new ScrollIterable(client, searchRequestBuilder).stream()
+                                                          Optional<Long> limit) {
+        SearchRequest searchRequest = getSearchRequest(mailboxIds, query, limit);
+        Stream<MessageSearchIndex.SearchResult> pairStream = new ScrollIterable(client, searchRequest).stream()
             .flatMap(this::transformResponseToUidStream);
 
         return limit.map(pairStream::limit)
             .orElse(pairStream);
     }
 
-    private SearchRequestBuilder getSearchRequestBuilder(Client client, Collection<MailboxId> users,
-                                                         SearchQuery query, Optional<Long> limit) {
+    private SearchRequest getSearchRequest(Collection<MailboxId> users, SearchQuery query, Optional<Long> limit) {
         return query.getSorts()
             .stream()
             .reduce(
-                client.prepareSearch(aliasName.getValue())
-                    .setTypes(typeName.getValue())
-                    .setScroll(TIMEOUT)
-                    .addFields(JsonMessageConstants.UID, JsonMessageConstants.MAILBOX_ID, JsonMessageConstants.MESSAGE_ID)
-                    .setQuery(queryConverter.from(users, query))
-                    .setSize(computeRequiredSize(limit)),
-                (searchBuilder, sort) -> searchBuilder.addSort(SortConverter.convertSort(sort)),
+                new SearchRequest(aliasName.getValue())
+                    .scroll(TIMEOUT),
+                (searchRequest, sort) -> searchRequest.source(getSearchSourceBuilder(users, query, limit, SortConverter.convertSort(sort))),
                 (partialResult1, partialResult2) -> partialResult1);
     }
 
+    private SearchSourceBuilder getSearchSourceBuilder(Collection<MailboxId> users, SearchQuery query, Optional<Long> limit, SortBuilder sort) {
```

Comment:
`SearchSourceBuilder` makes me think about searching on `_source` field? Is that what we want? or we want to search on the document?

Reply:
`SearchSourceBuilder` is part of the `SearchRequest` in ES6. That's where you can define the query, precise a sort, the size of the result, etc. We need a query for the search and this is the official way !

Reply:
It's not related to the `_source` ticket